web/src/api/images/get_uploaded_image.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as checkUserInAppDocumentSDK } from "@/api/images/graphql/checkUserInApp.generated";
+import { Session, getSession, withApiAuthRequired } from "@auth0/nextjs-auth0";
+import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+import { logger } from "@/lib/logger";
+
+export type ImageGetResponse = { url?: string; message?: string };
+
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+  image_type: yup
+    .string()
+    .strict()
+    .oneOf([
+      "logo_img",
+      "hero_image",
+      "showcase_img_1",
+      "showcase_img_2",
+      "showcase_img_3",
+    ])
+    .required(),
+});
+
+export type ImageGetBody = yup.InferType<typeof schema>;
+// This endpoint takes in an appID and image and returns that particular image
+export const handleImageGet = withApiAuthRequired(
+  async (req: NextApiRequest, res: NextApiResponse<ImageGetResponse>) => {
+    try {
+      if (!req.method || req.method !== "GET") {
+        return errorNotAllowed(req.method, res, req);
+      }
+      const { app_id, image_type } = req.query;
+
+      const { isValid, parsedParams, handleError } =
+        await validateRequestSchema({
+          value: { app_id, image_type },
+          schema,
+        });
+
+      if (!isValid || !parsedParams) {
+        return handleError(req, res);
+      }
+      const session = (await getSession(req, res)) as Session;
+      const auth0Team = session?.user.hasura.team_id;
+
+      const client = await getAPIServiceGraphqlClient();
+      const { team: userTeam } = await checkUserInAppDocumentSDK(
+        client
+      ).CheckUserInApp({
+        team_id: auth0Team,
+        app_id: app_id as string,
+      });
+      if (!userTeam[0].apps.some((app) => app.id === app_id)) {
+        return errorHasuraQuery({
```

Comment:
if we're using these errors, then this API routes should be exposed through Hasura custom actions (which I think may be a good idea) instead of as direct API routes. thoughts @igorosip0v ?

Reply:
Mostly it will be just a wrapper, but this will give us the possibility to generate hooks to gql query and then use it on the frontend side, so we can do this.

Reply:
as aligned on stand up, let's convert these endpoints to GraphQL custom actions